main/test/ca/uwaterloo/flix/library/TestLazyList.flix

```diff
@@ -0,0 +1,519 @@
+use LazyList.LazyList;
+use LazyList.LazyList.{Empty, LazyCons};
+
+namespace TestLazyList {
+
+    /////////////////////////////////////////////////////////////////////////////
+    // repeat                                                                  //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def repeat01(): Bool =
+        LazyList.take(1, LazyList.repeat(1)) == LazyCons(1, lazy Empty)
```

Comment:
Hmm, so I must design the test cases to reduce to some non-lazy value?

Reply:
You saw that I have defined an __eq method which forces the lists, right? :)

Reply:
> Hmm, so I must design the test cases to reduce to some non-lazy value?

Yes

Reply:
> You saw that I have defined an __eq method which forces the lists, right? :)

OK, so that works now, but in the near future we will have a type class for Eq and we will *not* allow equality on lazy values. (This default behavior would be dangerous, e.g. if you have infinite lists). 

So I agree that your solution works now, but lets future-proof it immediately :)

Reply:
Alright, just wanted to make sure that it wouldn't work moving forward :)
I'll refactor.